app/src/main/java/com/infomaniak/mail/ui/main/thread/ThreadViewModel.kt

```diff
@@ -95,29 +118,139 @@ class ThreadViewModel @Inject constructor(
     fun reassignMessagesLive(threadUid: String) {
         messagesLiveJob?.cancel()
         messagesLiveJob = viewModelScope.launch(ioCoroutineContext) {
+            messageController.getSortedAndNotDeletedMessagesAsync(threadUid)
+                ?.map { mapRealmMessagesResult(it.list, threadUid) }
+                ?.collect(messagesLive::postValue)
+        }
+    }
 
-            val cachedSplitBodies = mutableMapOf<String, SplitBody>()
-
-            suspend fun splitBody(message: Message): Message = withContext(ioDispatcher) {
-                if (message.body == null) return@withContext message
+    private suspend fun mapRealmMessagesResult(
+        messages: RealmResults<Message>,
+        threadUid: String,
+    ): Pair<ThreadAdapterItems, MessagesWithoutHeavyData> {
+
+        val items = mutableListOf<Any>()
+        val messagesToFetch = mutableListOf<Message>()
+        val thread = messages.firstOrNull()?.threads?.firstOrNull { it.uid == threadUid } ?: return items to messagesToFetch
+        val firstIndexAfterBlock = computeFirstIndexAfterBlock(thread, messages)
+        shouldDisplaySuperCollapsedBlock = shouldSuperCollapsedBlockBeDisplayed(messages.count(), firstIndexAfterBlock)
+        val shouldCreateSuperCollapsedBlock = shouldSuperCollapsedBlockBeCreated()
```

Comment:
The method `shouldSuperCollapsedBlockBeCreated` has a side effect by instantiating an empty mutable map inside a function that should only return a boolean